File: app/src/main/java/com/konektedi/vs/home/candidates/CandidatesViewModel.java
package com.konektedi.vs.home.candidates;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import com.konektedi.vs.utilities.NetworkStatus;

import java.util.List;


public class CandidatesViewModel extends AndroidViewModel {

    private LiveData<List<CandidatesModel>> categories;
    private LiveData<NetworkStatus> networkStatus;
    private CandidatesRepository repository = new CandidatesRepository();


    public CandidatesViewModel(@NonNull Application application) {
        super(application);
        networkStatus = repository.networkState;

    }

    public LiveData<List<CandidatesModel>> getAllCandidates(String election_id,String category_id) {
        categories = repository.getCandidates(election_id,category_id);
        return categories;
    }

    LiveData<NetworkStatus> getNetworkStatus() {
        return networkStatus;
    }


}

File: app/src/main/java/com/konektedi/vs/home/candidates/CandidatesViewModel.kt
package com.konektedi.vs.home.candidates

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData
import android.arch.lifecycle.ViewModel
import com.konektedi.vs.utilities.common.NetworkState
import com.konektedi.vs.utilities.models.Candidate


class CandidatesViewModel : ViewModel() {
    private val repository = CandidatesRepository()

    val networkState = repository.mNetworkState

    fun getCandidates(election_id: Int, category_id: Int): LiveData<List<Candidate>>? {
        return repository.getCandidates(election_id, category_id)
    }

    fun submitVote(map: Map<String, String>): MutableLiveData<NetworkState> {
        return repository.vote(map)
    }
}




